engine/src/flutter/impeller/renderer/pipeline_descriptor.cc

```diff
@@ -291,4 +292,14 @@ const std::vector<Scalar>& PipelineDescriptor::GetSpecializationConstants()
   return specialization_constants_;
 }
 
+void PipelineDescriptor::SetBasePipeline(
```

Comment:
Instead of maintaining back pointers and such. I wonder if it just makes sense to log the use counts of each variant. The dump can then just return the entire set. We'll be able to tell which variants were used how many times. If needed, we can merge them after reading.